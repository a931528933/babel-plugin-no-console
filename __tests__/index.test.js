// src/__tests__/index-test.js
const babel = require('@babel/core');
const plugin = require('../build/index.js').default;

it('条件语句块', () => {
  const example = `
  const foo = 1;
  if (foo) console.log(foo);
  `
  const {code} = babel.transform(example, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});

it('函数内部', () => {
  const example = `
  const foo = 1;
  function aaa(){
    console.log(foo)
  };
  `
  const {code} = babel.transform(example, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});

it('多层函数内部', () => {
  const example = `
  const foo = 1;
  function aaa(){
    function aaa(){
      console.log(foo)
    };
  };
  `
  const {code} = babel.transform(example, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});

it('类方法内部', () => {
  const example = `
  class A{
    getData(){
      console.log(1111)
    }
  }
  `
  const {code} = babel.transform(example, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});
