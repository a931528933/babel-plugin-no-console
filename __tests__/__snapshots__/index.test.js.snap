// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`函数内部 1`] = `
"const foo = 1;
function aaa() {}
;"
`;

exports[`多层函数内部 1`] = `
"const foo = 1;
function aaa() {
  function aaa() {}
  ;
}
;"
`;

exports[`条件语句块 1`] = `
"const foo = 1;
if (foo) {}"
`;

exports[`类方法内部 1`] = `
"class A {
  getData() {}
}"
`;
